### PR TITLE
fix(docs):  Upgrade `is-terminal` to Fix #96

### LINKED ISSUE #96: [Feature Request] Upgrade to `is-terminal` v0.4.6 or later
Due to an internal compiler error in `rustc` v1.69.0, Clap v4.1.8 and later cannot be documented anymore due a redundant documentation comment in `rustix` v0.36.9 which was propagated by `is-terminal` v0.4.4 directly to Clap.  The `rustix` team already works on a fix for the v0.36.x series of their crate, see https://github.com/bytecodealliance/rustix/pull/627, but the more current v0.37.x series already has the problem fixed.

Your direct dependency `is-terminal` applies a fixed version of `rustix` beginning with v0.4.6, see https://github.com/sunfishcode/is-terminal/pull/20 and https://github.com/sunfishcode/is-terminal/issues/22 for further details.  I would like to ask you to consider an upgrade to at least `is-terminal` v0.4.6 in order to fix documentation problems of Clap v4.1.8 and later with Rust v1.69.0.  This would also solve https://github.com/clap-rs/clap/discussions/4855.

### PR DESCRIPTION
**Pull request by kevinmatthes**
This commit upgrades `is-terminal` to v0.4.6 in order to fix #96.

Furthermore, the code was optimised by `cargo clippy --fix` as well as `cargo fmt`.  All tests succeed, the crate still compiles.  The code coverage is 44.45%.

### CODE REVIEW

File: crates/anstyle-parse/src/state/definitions.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use core::mem;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 #[repr(u8)]
+#[derive(Default)]
```
Comment (epage) [inline crates/anstyle-parse/src/state/definitions.rs:5]: This commit and PR do too much.  The clippy changes are unrelated to the version req changes.  Please revert this

File: crates/anstream/Cargo.toml [hunk 2]
```diff
@@ -34,7 +34,7 @@ anstyle = { version = "1.0.0", path = "../anstyle" }
 anstyle-parse = { version = "0.2.0", path = "../anstyle-parse" }
 colorchoice = { version = "1.0.0", path = "../colorchoice", optional = true }
 anstyle-query = { version = "1.0.0", path = "../anstyle-query", optional = true }
-is-terminal = { version = "0.4.4", optional = true }
+is-terminal = { version = "0.4.6", optional = true }
```
Comment (epage) [inline crates/anstream/Cargo.toml:37]: >  I would like to ask you to consider an upgrade to at least is-terminal v0.4.6 in order to fix documentation problems of Clap v4.1.8 and later with Rust v1.69.0.

Note: changing this crate is not needed for changing your crate.  You can run `cargo update -p is-terminal` and get the fix

Comment (epage) [reply crates/anstream/Cargo.toml:37]: (this does ensure *we* can build our own documentation when we next release)

**Comment by kevinmatthes** [comment]
Thank you for your feedback, @epage!

Your advice with `cargo update -p is-terminal` solves the problem for me, I already added this as solution to the discussion I opened in Clap, see https://github.com/clap-rs/clap/discussions/4855, such that other users with the same problem can find an answer there.  I will thus now close this PR.

File: crates/anstyle-parse/src/state/definitions.rs (hunk 1 above)
Comment (kevinmatthes) [reply crates/anstyle-parse/src/state/definitions.rs:5]: Okay, I am sorry.  I run all usual Cargo checks to ensure that I did not break something and Clippy suggested these changes to me.  I thought it would be useful to apply these suggestions.

File: crates/anstream/Cargo.toml (hunk 2 above)
Comment (kevinmatthes) [reply crates/anstream/Cargo.toml:37]: Thank you for your help!  Your hint solved the problem for me, so I withdrew this PR.  For documentation, I added some information how to solve the problem to a Clap discussion, see https://github.com/clap-rs/clap/discussions/4855.
